lib/assessment.js

```diff
@@ -8,38 +8,64 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-var min = function min() {
-
+var min = function min(previousValue, currentValue) {
+  return previousValue < currentValue ? previousValue : currentValue;
 };
-
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
-var sum = function sum() {
+var sum = [[0,1,2,3].reduce(function sum(a,b) {
+return a + b})];
 
-};
 
 // Question 3
 // Write a function that returns true
 // if invoking the callback for any element
 // in the array returns a truthy value
 var any = function any(array, callback, thisArg) {
-
-};
+  var result = false;
+    array.forEach(function(currentValue, index, array) {
+      result = result || callback(currentValue, index, array);
+    });
+      return result;
+    };
 
 // Question 4
 // Write a function that returns a new array
 // made up of the array elements for which invoking
 // the callback returns a truthy value
 var select = function select(array, callback, thisArg) {
+  var newArray = [];
+  var innerCallback = function(currentValue, index, array) {
+
+    newArray.push(callback(currentValue, index, array));
+  };
+
+  array.forEach(innerCallback);
+
+
+
+  return newArray;
+};
```

Comment:
This is not a filter function, but a map function. It will produce a new array containing the booleans returned by `callback` for the elements in the original array.

A filter function uses the `callback`'s return values to determine whether to add the original array element to the new array; it does not store the booleans anywhere.

1/2